### ISSUE TITLE
Coefficients in a small quadratic field, but not shown explicitly

### ISSUE THREAD

**Issue by davidfarmer** [LMFDB/lmfdb#2997]
The arithmetically normalized coefficients of this L-function
   http://beta.lmfdb.org/L/ModularForm/GL2/Q/holomorphic/1041/2/a/a/1/1/
lie in Qsqrt5.  But on the L-function home page those are only shown
as decimals.  It would be nice to see those coefficients explicitly.

This is an interesting example, because it has rank 2, but not rational integer
coefficients.  And its Galois conjugate also has rank 2:
   http://www.lmfdb.org/L/ModularForm/GL2/Q/holomorphic/1041/2/a/a/1/2/
which apparently follows from some conjecture.

**Comment by AndrewVSutherland** [comment]
Hmm, currently I don't think the L-function "knows" the (embedded) number field in which its coefficients lie, if any (although we could and perhaps should add this information to the database in order to make the L-functions fully independent of their origins).

Are you saying you would like to see arithmetic L-functions store their coefficients as exact values (which in general would mean storing them as algebraic expressions along with a choice of embedding) that would be displayed when the arithmetic normalization is chosen?

In the real quadratic case there is an unambiguous way to display the  image of a generator for the coefficient field under a chosen embedding (e.g. \sqrt{5} denotes the positive square root of 5), but in general this is going to be a bit tricky -- if the coefficients lie in, say, one of the embeddings of the cubic field Q[x]/(x^3+x+1), how would you want them displayed?

Or are you suggesting we treat real (and possibly also imaginary) quadratic fields as a special case?

**Comment by edgarcosta** [comment]
I think it would be tricky and overwhelming to display the coefficients as algebraic integers. 
The first question is, what do we gain with it?
Second, will it be distracting?
We went through a lot of hoops to get the q-expansions for classical modular forms to display in some reasonable manner by picking an optimal basis.
Doing something like that for L-fcns might be a bit distracting from some of the key content, e.g., the zeros and the euler factors.

**Comment by AndrewVSutherland** [comment]
@edgarcosta I agree with you, although I can imagine making an exception for quadratic fields (which is perhaps all that @davidfarmer is suggesting).

However, this issue does raise a point that I think we should consider.  An arithmetic L-function really ought to "know" the embedded number field in which its Dirichlet coefficients lie, even if we don't use this information when displaying them.  This is an important invariant of the L-function.

**Comment by davidfarmer** [comment]
I agree that the field of coefficients is an important invariant
of an arithmetic L-function and it should (eventually) be somewhere
on the home page.

Note that Maass type L-functions don't have an associated number
field, and in fact should not have the option of toggling between
arithmetic and analytic normalizations.  We can deal with that
when the GL(2) Maass form L-functions are computed rigorously.

I was suggesting that the case of coefficients in a quadratic
field be given special treatment.  (Although I am not sure
why seeing q-expansion coefficients explicitly is important
but seeing Dirichlet coefficients is not.)

**Comment by edgarcosta** [comment]
I agree that we should store the field as an invariant.

However, I would like to point out that this L-fcn corresponds to the embedded modular form: http://beta.lmfdb.org/ModularForm/GL2/Q/holomorphic/1041/2/a/a/1/1/
and for that one, we also only display the coefficients as floating point numbers, and one needs to navigate to the Galois orbit to see them as algebraic integers.

**Comment by AndrewVSutherland** [comment]
@davidfarmer Regarding your last point, on embedded newform pages (which are the objects that give rise to the L-functions we are considering) we do not display algebraic q-expansions.  We only display algebraic q-expansions on the page for the Galois orbit of the newform, and there we are not picking an embedding.  For example, if you go to

http://www.lmfdb.org/L/ModularForm/GL2/Q/holomorphic/53/2/b/a/52/1/

you will see that the q-expansion coefficients are the same (but with more precision) as those listed in the Dirihlet series of its L-function

http://www.lmfdb.org/L/ModularForm/GL2/Q/holomorphic/53/2/b/a/52/1/

It is only the the home page for the Galois orbit of the modular form that we display an algebraic q-expansion

http://www.lmfdb.org/ModularForm/GL2/Q/holomorphic/53/2/b/a/

This q-expansion does not depend on a choice of embedding,, where as the embedded modular forms and their L-functions do.

**Comment by davidfarmer** [comment]
Right.  Those pages might also be improved by showing the quadratic case 
explicitly.

I'm not suggesting this is a 1.1 priority.  And I think it is more
relevant to L-functions, where one could toggle between arithmetic
(explicit, showing the radicals) and analytic (decimals).

**Comment by roed314** [comment]
Reading through the conversation, here's what I think we've agreed to do (time permitting):

1. Add the field of coefficients for arithmetic L-functions
2. Remove the ability to toggle to an arithmetic normalization for Maass type L-functions.

Since these aren't suggested in the main text of the issue, should they have their own issues?

I don't see a consensus on the original suggestion.  @davidfarmer, @AndrewVSutherland, @edgarcosta, what do you think now?

**Comment by AndrewVSutherland** [comment]
I think we should create separate issues for the two items you listed and close this one.  I do not think we should try to display Dirichlet coefficients as algebraic numbers, even when they are quadratic; I think this would make sense if we had pages for Galois orbits of L-functions (as we do for CMFs), but we don't and I'm not sure we should (if we did have such pages, they would need to be different from our current L-function pages -- we would need to disable the analytic normalization and change a bunch of other things).

**Comment by edgarcosta** [comment]
I agree with @AndrewVSutherland suggestions
